### PR TITLE
#1474 - Dynamic resizing string feature

### PR DESCRIPTION
**Pull request by wpmp**
**What's in the PR**
- add dynamic textarea feature editor

**How to test manually**
* Create a new string feature
* Enable multiple rows
* Enable dynamic size, see if collapsed and expanded rows disappears
* Check if the new feature resize in the annotation sidebar when filled

**Automatic testing**
* [ ] PR includes unit tests

**Documentation**
* [x] PR updates documentation

### CODE REVIEW

**Comment by reckart** [comment]
Jenkins, can you test this please?

**Comment by reckart** [comment]
Jenkins, can you test this please?

File: webanno-api-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/annotation/feature/editor/DynamicTextAreaFeatureEditor.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#
+#Unless required by applicable law or agreed to in writing, software
+#distributed under the License is distributed on an "AS IS" BASIS,
+#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#See the License for the specific language governing permissions and
+#limitations under the License.
+-->
+<html xmlns:wicket="http://wicket.apache.org">
+<wicket:panel>
+  <div class="form-group" wicket:enclosure="value">
```
[the remark this answers is not in the document]
Comment (wpmp) [reply webanno-api-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/annotation/feature/editor/DynamicTextAreaFeatureEditor.html]: Yes, I already noticed a problem when there is such a feature editor on the left and right sidebar at the same time.

**Comment by wpmp** [comment]
> Would it be sensible to allow setting a maximum number of rows such that scrolling kicks in when the text is longer than the maximum?

I think this would be a bit more difficult as the size of the textarea is calculated by the pixels of the scrollbar and not on the amount of rows there are.
I will have a look at this.

> Is it intentional that the field always appears to be one line larger than what seems to be necessary?

The problem is that some extra pixels need to be added to the textarea to completely remove the scrollbar.
I can check how much I can reduce them for Chrome but I don't know if this will work out the same way in Safari.

**Comment by reckart** [comment]
If you want to remove the scrollbar, doesn't it work to turn it off completely?

**Comment by wpmp** [comment]
Hm, didn't think about that..

**Comment by wpmp** [comment]
@reckart ready for another review. Regarding row limits could we maybe postpone this since we already have a limited textarea and I think it is going to be more difficult?

**Comment by reckart** [comment]
The JS code is not added multiple times anymore, but with multiple dynamically-sized editors, it is called for each editor and every times applies to **all** editors. If it is called multiple times, then each call should only apply to the respective editor, selecting it via its Wicket markup id.

**Comment by reckart** [comment]
Jenkins, can you test this please?

**Comment by wpmp** [comment]
@reckart this PR is ready for a review

**Comment by wpmp** [comment]
Jenkins, can you test this please?

**Comment by reckart** [comment]
Jenkins, can you test this please?

**Comment by reckart** [comment]
Jenkins, can you test this please?

**Pull request merged**